docs/spec/relayer/RelayerDefinitions.tla

```diff
@@ -90,111 +297,178 @@ ConnectionEnds ==
     - connectionEnd : a connection end record 
       Stores data about the connection with the counterparty chain
  ***************************************************************************)
-Chains ==    
+ChainStores(maxHeight, channelOrdering, maxPacketSeq) ==    
     [
-        height : Nat,
-        counterpartyClientHeights : SUBSET(Nat),
-        connectionEnd : ConnectionEnds
-    ]
+        height : 1..maxHeight,
+        counterpartyClientHeights : SUBSET(1..maxHeight),
+        connectionEnd : ConnectionEnds(channelOrdering, maxPacketSeq),
+        packetCommitment : SUBSET(PacketCommitments(maxHeight, maxPacketSeq))
+    ] <: {ChainStoreType}
+
+(********************************* Packets *********************************
+ A set of packets.
+ ***************************************************************************)
+Packets(maxHeight, maxPacketSeq) ==
+    [
+        sequence : 1..maxPacketSeq,
+        timeoutHeight : 1..maxHeight,
+        srcChannelID : ChannelIDs,
+        dstChannelID : ChannelIDs
+    ] <: {PacketType}
 
 (******************************** Datagrams ********************************
  A set of datagrams.
  ***************************************************************************)
-Datagrams(Heights) ==
-    [type : {"CreateClient"}, clientID : ClientIDs, height : Heights]
+Datagrams(maxHeight, maxPacketSeq) ==
```

Comment:
This is impressive! But I also interpret it as a clear sign we should not grow this spec any more.